services/provider/server/server.go

```diff
@@ -1124,6 +1125,134 @@ func (s *OCSProviderServer) RequestMaintenanceMode(ctx context.Context, req *pb.
+	response := &pb.StorageClientsInfoResponse{}
+	for i := range req.ClientIDs {
+		consumer, err := s.consumerManager.GetByClientID(ctx, req.ClientIDs[i])
+		if err != nil {
+			klog.Errorf("failed to get consumer with client id %v: %v", req.ClientIDs[i], err)
+			response.Errors = append(response.Errors,
+				&pb.StorageClientInfoError{
+					ClientID: req.ClientIDs[i],
+					Code:     pb.ErrorCode_Internal,
+					Message:  err.Error(),
+				},
+			)
+		}
+		if consumer == nil {
+			klog.Infof("no consumer found with id %v", req.ClientIDs[i])
+			continue
+		}
+
+		owner := util.FindOwnerRefByKind(consumer, "StorageCluster")
+		if owner == nil {
+			klog.Infof("no owner found for consumer %v", req.ClientIDs[i])
+			continue
+		}
+
+		if owner.UID != types.UID(req.StorageClusterUID) {
+			klog.Infof("storageCluster specified on the req does not own the client %v", req.ClientIDs[i])
+			continue
+		}
+
+		rnsList := &rookCephv1.CephBlockPoolRadosNamespaceList{}
+		err = s.client.List(
+			ctx,
+			rnsList,
+			client.InNamespace(s.namespace),
+			client.MatchingLabels{controllers.StorageConsumerNameLabel: consumer.Name},
+			client.Limit(2),
+		)
+		if err != nil {
+			response.Errors = append(response.Errors,
+				&pb.StorageClientInfoError{
+					ClientID: req.ClientIDs[i],
+					Code:     pb.ErrorCode_Internal,
+					Message:  "failed to get radosnamespaces",
+				},
+			)
+			klog.Error(err)
+			continue
+		}
+		if len(rnsList.Items) > 1 {
+			response.Errors = append(response.Errors,
+				&pb.StorageClientInfoError{
+					ClientID: req.ClientIDs[i],
+					Code:     pb.ErrorCode_Internal,
+					Message:  "invalid number of radosnamespace found for the Client",
+				},
+			)
+			klog.Errorf("invalid number of radosnamespace found for the Client %v", req.ClientIDs[i])
+			continue
+		}
+		clientInfo := &pb.ClientInfo{ClientID: req.ClientIDs[i]}
+		if len(rnsList.Items) == 1 {
+			clientInfo.RadosNamespace = rnsList.Items[0].Name
+		}
+		response.ClientsInfo = append(response.ClientsInfo, clientInfo)
+	}
+
+	return response, nil
+}
+
+func (s *OCSProviderServer) GetBlockPoolsInfo(ctx context.Context, req *pb.BlockPoolsInfoRequest) (*pb.BlockPoolsInfoResponse, error) {
+	response := &pb.BlockPoolsInfoResponse{}
+
+	for i := range req.BlockPoolNames {
+		cephBlockPool := &rookCephv1.CephBlockPool{}
+		cephBlockPool.Name = req.BlockPoolNames[i]
+		cephBlockPool.Namespace = s.namespace
+		err := s.client.Get(ctx, client.ObjectKeyFromObject(cephBlockPool), cephBlockPool)
+		if kerrors.IsNotFound(err) {
+			klog.Infof("blockpool %v not found", cephBlockPool.Name)
+			continue
+		} else if err != nil {
+			klog.Errorf("failed to get blockpool %v: %v", cephBlockPool.Name, err)
+			response.Errors = append(response.Errors,
+				&pb.BlockPoolInfoError{
+					BlockPoolName: cephBlockPool.Name,
+					Code:          pb.ErrorCode_Internal,
+					Message:       err.Error(),
```

Comment:
Let's keep the internal of the issue on the provider side
```suggestion
					Message:       "failed loading block pool information",
```